Soccar [Main-Server]/src/nl/soccar/mainserver/rmi/MainServerController.java

```diff
@@ -73,12 +81,54 @@ public MainServerController() {
         } catch (RemoteException e) {
             LOGGER.log(Level.SEVERE, "An error occurred while locating and/or binding the registry.", e);
         }
+
+        continuouslyPingGameServers();
+    }
+
+    /**
+     * Continously pings all game servers to check if the RMI connection is
+     * active. A random value is sent to the game servers for them to process
+     * with a pre-defined formula. If the result of the formula is not correct,
+     * the connected game server is possibly rogue. If the result of the formula
+     * is not correct or there was no response at all, the game server is
+     * deregistered.
+     */
+    public void continuouslyPingGameServers() {
+        timer.scheduleAtFixedRate(new TimerTask() {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 27 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1188)